### DISCUSSION TITLE
Image export

### DISCUSSION THREAD

**Discussion by spd-spd** [RobertWeinmeister/dokuwiki-mermaid#10] [Ideas]
[Windows 10 Dokuwiki-on-a-stick 2023-04-04a "Jack Jackrum" plugin:mermaid 10.2c 2023/7/24]
I'm looking to use this plugin to generate diagrams, and it does them reasonably well.  However I need some way to export them to a file in png or svg format, perhaps even save them back into the wiki as alternate display if the render fails or for other plugins like dw2pdf that can't render correctly on server side.  Currently I don't see a way to export or save the generated diagram other than a screenshot, which for large renders means 2 or 3 images that I have to stitch together again afterwards.

**Comment by RobertWeinmeister** [comment]
Mermaid itself does not support an export, but there are two other options. If it is not too many files or too often, https://mermaid.live/ allows the export to various formats with some copy and paste. Otherwise there is a command-line interface, https://github.com/mermaid-js/mermaid-cli.

**Reply by spd-spd** [reply]
I was expecting something similar to the command line interface code in the wiki webpage would work.  I had a look at the code in the repository and doesn't look like there is much coding to do it.  Unfortunately my .js coding skills aren't up to that level yet to say whether it would be easy or not.  And licensing is another story, not sure how well gpl and mit ones go together.

**Comment by RobertWeinmeister** [comment]
The newest release allows this: [Exporting Diagrams](https://github.com/RobertWeinmeister/dokuwiki-mermaid?tab=readme-ov-file#-exporting-diagrams)
